sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/CallingServerAsyncClient.java

```diff
@@ -0,0 +1,418 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.communication.callingserver;
+
+import com.azure.communication.callingserver.implementation.AzureCommunicationCallingServerServiceImpl;
+import com.azure.communication.callingserver.implementation.CallConnectionsImpl;
+import com.azure.communication.callingserver.implementation.ServerCallsImpl;
+import com.azure.communication.callingserver.implementation.converters.CallConnectionRequestConverter;
+import com.azure.communication.callingserver.implementation.converters.JoinCallConverter;
+import com.azure.communication.callingserver.implementation.converters.CallingServerErrorConverter;
+import com.azure.communication.callingserver.implementation.models.CommunicationErrorException;
+import com.azure.communication.callingserver.implementation.models.CreateCallRequestInternal;
+import com.azure.communication.callingserver.models.CreateCallOptions;
+import com.azure.communication.callingserver.models.JoinCallOptions;
+import com.azure.communication.callingserver.models.ParallelDownloadOptions;
+import com.azure.communication.common.CommunicationIdentifier;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.http.HttpRange;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.SimpleResponse;
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousFileChannel;
+import java.nio.file.OpenOption;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import static com.azure.core.util.FluxUtil.fluxError;
+import static com.azure.core.util.FluxUtil.monoError;
+import static com.azure.core.util.FluxUtil.withContext;
+
+
+/**
+ * Async client that supports server call operations.
+ */
+@ServiceClient(builder = CallingServerClientBuilder.class, isAsync = true)
+public final class CallingServerAsyncClient {
+    private final CallConnectionsImpl callConnectionInternal;
+    private final ServerCallsImpl serverCallInternal;
+    private final ClientLogger logger = new ClientLogger(CallingServerAsyncClient.class);
+    private final ContentDownloader contentDownloader;
+
+    CallingServerAsyncClient(AzureCommunicationCallingServerServiceImpl callServiceClient) {
+        callConnectionInternal = callServiceClient.getCallConnections();
+        serverCallInternal = callServiceClient.getServerCalls();
+
+        contentDownloader = new ContentDownloader(
+            callServiceClient.getEndpoint(),
+            callServiceClient.getHttpPipeline(),
+            logger);
```

Comment:
Don't pass in logger. By convention, each class has its own logger unless you have good reasons